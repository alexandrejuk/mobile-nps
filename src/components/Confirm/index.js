import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './style.module.css'

const Confirm = ({
  children,
  close,
  show,
}) => (
  <div
    onClick={close}
    className={classNames(styles.bgConfirm, {
      [styles.show]: show,
    })}
  >
  <div
    className={classNames(styles.confirm, {
      [styles.show]: show,
    })}
  >
      {children}
    </div>
   </div>
)

Confirm.propTypes = {
  children: PropTypes.node.isRequired,
  close: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
}

export default Confirm
