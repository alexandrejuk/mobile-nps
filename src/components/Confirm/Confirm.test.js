import React from 'react'
import {
  fireEvent,
  render,
} from '@testing-library/react'

import Confirm from '.'

const MODAL_TEXT = 'I am a modal component!'
describe('Modal component', () => {
  it('should render correctly', () => {
    const callback = jest.fn()
    const { container, getByText } = render(
      <Confirm
        show
        close={callback}
      >
        {MODAL_TEXT}
      </Confirm>
    )
    const SampleModal = getByText(MODAL_TEXT)
    expect(callback).not.toHaveBeenCalled()
    fireEvent.click(SampleModal)
    expect(callback).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })
})
