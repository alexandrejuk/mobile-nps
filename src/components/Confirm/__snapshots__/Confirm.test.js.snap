// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render correctly 1`] = `
<div>
  <div
    class="bgConfirm show"
  >
    <div
      class="confirm show"
    >
      I am a modal component!
    </div>
  </div>
</div>
`;
